archivessnake/create_daos.py

```diff
@@ -0,0 +1,190 @@
+                ao_uri, len(dip)))
+
+
+def get_dip_info(dip):
+    """Parses DIP information to return DIP UUID, title to be used as DAO title, and UUID of associated AIP.
+
+    Args:
+        dip (dictionary): DIP information
+
+    Returns:
+        UUID, file title, and associated AIP UUID (tuple)
+    """
+    dip_uuid = dip['uuid']
+    title = dip['current_full_path'].split('/')[-1][:-37]
+    aip_uuid = dip['related_packages'][0].split('/')[-2]
+    return dip_uuid, title, aip_uuid
+
+
+def post_dao(aspace, dip_uuid, title, aip_uuid):
+    """Creates a digital object in ArchivesSpace
+
+    Args:
+        aspace (asnake.aspace.ASpace): instantiation of ASpace
+        dip_uuid (str): DIP UUID
+        title (str): DAO title
+        aip_uuid (str): AIP UUID
+
+    Returns:
+        DAO URI (string)
+    """
+    file_uri = join("http://storage.rockarch.org/",
+                    "{}-{}.pdf".format(dip_uuid, title))
+    data = {"jsonmodel_type": "digital_object",
+            "is_slug_auto": True,
+            "file_versions": [{"jsonmodel_type": "file_version",
+                               "is_representative": False,
+                               "file_uri": str(file_uri),
+                               "xlink_actuate_attribute": "onRequest",
+                               "xlink_show_attribute": "new",
+                               "file_format_name": "pdf",
+                               "file_format_version": "Generic PDF",
+                               "publish": True}],
+            "restrictions": False,
+            "notes": [{"jsonmodel_type": "note_digital_object",
+                       "content": [str(aip_uuid)],
+                       "type": "originalsloc",
+                       "publish": False}],
+            "title": str(title),
+            "digital_object_id": str(dip_uuid)}
+    r = aspace.post("repositories/2/digital_objects", json=data)
+    if r.status_code != 200:
+        raise Exception("Could not post {}: {}".format(
+            title, r.json().get('error')))
+    else:
+        return r.json().get("uri")
+
+
+def update_component(aspace, component, dao):
+    """Add a digital object to an archival object
+
+    Args:
+        aspace (asnake.aspace.ASpace): instantiation of ASpace
+        component (str): ArchivesSpace URI for archival object
+        dao (str): ArchivesSpace URI for digital object
+
+    Returns:
+        Response from ArchivesSpace
+    """
+    updated_component = aspace.client.get(component).json()
+    updated_component["instances"].append(
+        {
+            "instance_type": "digital_object",
+            "jsonmodel_type": "instance",
+            "digital_object": {
+                "ref": dao}})
+    r = aspace.client.post(component, json=updated_component)
+    if r.status_code != 200:
+        raise Exception("Could not post {} to {}: {}".format(
+            dao, component, r.json().get('error')))
+    else:
+        return r
```

Comment:
I am not sure it's necessary to return anything here, as you're not doing anything with it in `main`.